Lib/importlib/_bootstrap.py

```diff
@@ -51,18 +51,106 @@ def _new_module(name):
 
 # Module-level locking ########################################################
 
-# A dict mapping module names to weakrefs of _ModuleLock instances
-# Dictionary protected by the global import lock
+# For a list that can have a weakref to it.
+class _List(list):
+    pass
```

Comment:
FWIW, I think it would be nice to use __slots__ for this class, to reduce the size of each instance (especially since the setdefault call creates one each time, even when it isn't necessary).

Reply:
As in you're going to make a PR to make that change, or you're hoping someone else will open an issue on your behalf to track the idea?

Reply:
The latter (although not on _my_ behalf) -- I don't know if there's a reason not to do it, and I don't have the spare cycles to dig in and find out.